control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -668,6 +662,12 @@ func (r *Reconciler) reconcilerBrokerResource(ctx context.Context, topic string,
 	}
 	resource.EgressConfig = egressConfig
 
+	eventPolicies, err := coreconfig.EventPoliciesFromAppliedEventPoliciesStatus(broker.Status.AppliedEventPoliciesStatus, r.EventPolicyLister, broker.Namespace, features)
```

Comment:
it's weird we're relying on the status, so we require a precise ordering for operations, if this method `reconcilerBrokerResource` depends on having the applied event policies, I'd make it explicit with a parameter

Reply:
Same for other reconcilers

Reply:
I see what you mean. The problem is, that we don't have access at this point otherwise to the applying eventpolicies. `auth.UpdateStatusWithEventPolicies()` from eventing-core sets the applying (and ready) policies in the status and marks the conditions.

So I see two alternatives here:
1. use the `auth.GetEventPoliciesForResource()` to get the applying policies, filter on the ready ones and pass them to `reconcilerBrokerResource()` (similar to passing the audience). This has the drawback, that we need to load the eventpolicies anyhow later again (in `auth.UpdateStatusWithEventPolicies()`) as we need to mark the conditions. And this would kind of duplicate the logic from `auth.UpdateStatusWithEventPolicies()` to "filter" which policies apply for a resource
2. Refactor the `auth.UpdateStatusWithEventPolicies()` to take a list of event policies which apply for a resource. Anyhow, this would mean, that we need to duplicate the `auth.GetEventPoliciesForResource()` in each reconciler (in eventing-core too).

Objections or other ways?

Reply:
@pierDipi regarding 2: something like https://github.com/knative/eventing/pull/8173. Anyhow this would still mean, we need to duplicate the logic, which event policies apply at the end to a resource (only the ready ones) :/

Reply:
The difference is that the caller knows and it's orchestrating the order of operations and the expected order can be even be more explicit with a temporary variable

```golang
	appliedEventPolicy := ...
	err = auth.UpdateStatusWithEventPolicies(features, &appliedEventPolicy, &broker.Status, r.EventPolicyLister, eventing.SchemeGroupVersion.WithKind("Broker"), broker.ObjectMeta)
	if err != nil {
		return fmt.Errorf("could not update broker status with EventPolicies: %v", err)
	}

	// Get resource configuration.
	brokerResource, err := r.reconcilerBrokerResource(ctx, topic, broker, secret, topicConfig, audience, appliedEventPolicy)
	if err != nil {
		return statusConditionManager.FailedToResolveConfig(err)
	}

       broker.Status.AppliedEventPoliciesStatus = appliedEventPolicy

Reply:
Also I've noticed this

the function `UpdateStatusWithEventPolicies` resents the status to empty so if there is error on, for example `GetEventPoliciesForResource` we will have no policies applied but I would expect the system to remain configured the the latest known "good" states until it recovers

```
func UpdateStatusWithEventPolicies(featureFlags feature.Flags, status *eventingduckv1.AppliedEventPoliciesStatus, statusMarker EventPolicyStatusMarker, eventPolicyLister listerseventingv1alpha1.EventPolicyLister, gvk schema.GroupVersionKind, objectMeta metav1.ObjectMeta) error {
	status.Policies = nil
//... 
```

Reply:
Ah ok. Now I understand better what you mean.